Dockerfile

```diff
@@ -48,8 +47,17 @@ RUN chmod -R +x ${SALT_RUNTIME_DIR}
 
 COPY assets/sbin/* /usr/local/sbin/
 
+# Add python-ldap for ldap external_auth
+# FIXME: Downgrade to pip 22.3.1 for bug: https://github.com/saltstack/salt/issues/65025
+RUN salt-pip install "pip==22.3.1"
+RUN salt-pip install python-ldap
+
 # Cleaning tasks
 RUN rm -rf "${SALT_BUILD_DIR:?}"/*
+RUN DEBIAN_FRONTEND=noninteractive apt-get purge --yes --quiet gcc libsasl2-dev libldap2-dev \
+  && DEBIAN_FRONTEND=noninteractive apt-get --yes --quiet autoremove \
+  && DEBIAN_FRONTEND=noninteractive apt-get clean --yes \
+  && rm -rf /var/lib/apt/lists/*
```

Comment:
This is not necessary, as it is already done here:

https://github.com/cdalvaro/docker-salt-master/blob/44c73c718b652b94b180c9214acdb9c8417e04b1/assets/build/install.sh#L116-L119

But you can add the `autoremove` command.

Reply:
Just checked and `install.sh` has `apt purge --auto-remove` and the man pages say: `this option acts like running the autoremove command` so probably don't need an explicit autoremove as well.